crates/db_connection_pool/src/dbconnection/odbcconn.rs

```diff
@@ -173,6 +174,26 @@ where
         let params = params.iter().map(dyn_clone::clone).collect::<Vec<_>>();
         let secrets = Arc::clone(&self.params);
 
+        let cancellation_token = tokio_util::sync::CancellationToken::new();
+        let cloned_token = cancellation_token.clone();
+
+        tokio::spawn(async move {
+            let ctrl_c = async {
+                let signal_result = signal::ctrl_c().await;
```

Comment:
Does this also handle SIGTERM? That is the signal K8s sends for graceful shutdown

Reply:
it seems spiced doesn't handle sigterm gracefully now. Created a separate issue to track this https://github.com/spiceai/spiceai/issues/2159